src/cargo/util/context/mod.rs

```diff
@@ -2348,6 +2360,16 @@ impl ConfigValue {
     }
 }
 
+/// List of which configuration lists cannot be merged.
+/// Instead of merging, these the higher priority list replaces the lower priority list.
+fn is_nonmergable_list(key: &ConfigKey) -> bool {
```

Comment:
I'm slightly concerned with this getting out of sync. As a small mitigation, can you add a bold sentence to the `PathAndArgs` type that says if you use it in a new key that this needs to be udpated?